packages/backend/src/apps/toolbelt/actions/if-then/index.ts

```diff
@@ -0,0 +1,387 @@
+
+      // Field is hidden; later PR will hide for real.
+      required: false,
+      variables: false,
+    },
+    {
+      // Stores the step ID to skip to if the branch is not taken. This is
+      // computed when the pipe is published. If the flow should terminate, then
+      // is just the empty string.
+      key: 'nextStepIdIfSkipped',
+      type: 'string' as const,
+      label: 'FILE A BUG IF YOU SEE THIS',
+
+      // Field is hidden; later PR will hide for real.
+      required: false,
+      variables: false,
+    },
+    {
+      label: 'Conditions',
+      key: 'conditions',
+      type: 'multirow' as const,
+      required: true,
+      description:
+        'Every condition has to be satisfied for this branch to be taken.',
+      subFields: [
+        {
+          placeholder: 'Field',
+          key: 'field',
+          type: 'string' as const,
+          required: true,
+          variables: true,
+        },
+        {
+          placeholder: 'Is or Not',
+          key: 'is',
+          type: 'dropdown' as const,
+          required: true,
+          variables: false,
+          options: [
+            { label: 'Is', value: 'is' },
+            { label: 'Not', value: 'not' },
+          ],
+        },
+        {
+          placeholder: 'Condition',
+          key: 'condition',
+          type: 'dropdown' as const,
+          required: true,
+          variables: false,
+          options: [
+            { label: '=', value: 'equals' },
+            { label: '>=', value: 'gte' },
+            { label: '>', value: 'gt' },
+            { label: '<=', value: 'lte' },
+            { label: '<', value: 'lt' },
+            { label: 'contains', value: 'contains' },
+          ],
+        },
+        {
+          placeholder: 'Value',
+          key: 'value',
+          type: 'string' as const,
+          required: true,
+          variables: true,
+        },
+      ],
+    },
+  ],
+
+  async testRun($) {
+    const isBranchTaken = shouldTakeBranch($)
+    $.setActionItem({
+      raw: { isBranchTaken },
+    })
+
+    if (isBranchTaken) {
+      return {}
+    }
+
+    // For test runs, nextStepIdIfSkipped is not populated yet, so we compute
+    // the next step manually.
+    const runningStep = await Step.query()
+      .findById($.step.id)
+      .withGraphFetched({ flow: { steps: true } })
+    const stepToSkipTo = getStepToSkipTo(runningStep)
+
+    return stepToSkipTo?.id
+      ? {
+          nextStep: {
+            command: 'jump-to-step',
+            stepId: stepToSkipTo.id,
+          },
+        }
+      : {
+          nextStep: { command: 'stop-execution' },
+        }
+  },
+
+  async run($) {
+    const isBranchTaken = shouldTakeBranch($)
+    $.setActionItem({
+      raw: { isBranchTaken },
+    })
+
+    if (isBranchTaken) {
+      return {}
+    }
+
+    const nextStepId = $.step.parameters.nextStepIdIfSkipped
+
+    if (nextStepId === null || nextStepId === undefined) {
```

Comment:
nit: if (!nextStepId)

Reply:
Actually for this, `nextStepId` is an empty string if there is no next branch to skip to.

How about `nextStepId == null` with comment saying that empty string is an expected value, hence this check